src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -198,24 +196,25 @@ export default class QueryEditorUI extends Component<Props, State> {
       language,
     };
 
-    const enhancement = this.props.settings.getQueryEnhancements(newQuery.language);
+    const enhancement = this.languageManager.getQueryEnhancements(newQuery.language);
     const fields = enhancement?.fields;
     const newSettings: DataSettings = {
       userQueryLanguage: newQuery.language,
       userQueryString: newQuery.query,
       ...(fields && { uiOverrides: { fields } }),
     };
-    this.props.settings?.updateSettings(newSettings);
-
-    const dateRangeEnhancement = enhancement?.searchBar?.dateRange;
-    const dateRange = dateRangeEnhancement
-      ? {
-          from: dateRangeEnhancement.initialFrom!,
-          to: dateRangeEnhancement.initialTo!,
-        }
-      : undefined;
-    this.onChange(newQuery, dateRange);
-    this.onSubmit(newQuery, dateRange);
+    this.languageManager.updateSettings(newSettings);
+
+    // TODO: confirm if this is needed
+    // const dateRangeEnhancement = enhancement?.searchBar?.dateRange;
+    // const dateRange = dateRangeEnhancement
+    //   ? {
+    //       from: dateRangeEnhancement.initialFrom!,
```

Comment:
i dont think we need this anymore. we did this initially because I thought quick date ranges like "Last 30 days" "Last 15 minutes" didn't work for PPL queries since I append the the date picker time filter to the query. but it should work with those quick ranges because we have a helper function to convert it

Reply:
or we can double down and have a `formatDateRange` handler when you register a language